### PR TITLE
REI Compatibility

### PR DESCRIPTION
**Pull request by lantice3720**
Installed with Roughly Enough Items mod, makes it easier to search items.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -50,6 +50,10 @@ dependencies {
 	// Mod Menu
 	modImplementation "com.terraformersmc:modmenu:${project.mod_menu_version}"
 
+	// REI
+	modImplementation "me.shedaniel:RoughlyEnoughItems-fabric:${project.rei_version}"
+	modRuntimeOnly "me.shedaniel:RoughlyEnoughItems-fabric:${project.rei_version}"
```
Comment (kevinthegreat1) [inline build.gradle:55]: This is what REI recommends, is there a reason why this is not used?
```
dependencies {
    modCompileOnly "me.shedaniel:RoughlyEnoughItems-api-fabric:VERSION"
    modRuntimeOnly "me.shedaniel:RoughlyEnoughItems-fabric:VERSION"
}
```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
I'm not too familiar with REI nor use it. So if you have tested this then it should be good.

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/itemlist/SkyblockCraftingRecipe.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SkyblockCraftingRecipe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SkyblockCraftingRecipe.class);
+    String craftText = "";
+    final List<ItemStack> grid = new ArrayList<>(9);
+    ItemStack result;
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/skyblock/itemlist/SkyblockCraftingRecipe.java:16]: Small nitpick.
```suggestion
    private String craftText = "";
    private final List<ItemStack> grid = new ArrayList<>(9);
    private ItemStack result;
```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Just want to confirm that you tested this. If that's the case, just a small nitpick and it looks good.

**Comment by Julienraptor01** [comment]
Is it automatically enabled if REI is installed and can it be manually disable with a setting ?

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/itemlist/SkyblockCraftingRecipe.java (hunk 2 above)
Comment (lantice3720) [reply src/main/java/me/xmrvizzy/skyblocker/skyblock/itemlist/SkyblockCraftingRecipe.java:16]: Thank you for reviewing. I've tested this a few times and it worked as intended.

**Comment by lantice3720** [comment]
> Is it automatically enabled if REI is installed and can it be manually disable with a setting ? 
Yes. this is automatically enabled if REI is installed.
iirc REI can be disabled easily via settings but I'm not sure whether only one plugin(features added by Skyblocker) can.

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/itemlist/SkyblockCraftingRecipe.java (hunk 2 above)
Comment (kevinthegreat1) [reply src/main/java/me/xmrvizzy/skyblocker/skyblock/itemlist/SkyblockCraftingRecipe.java:16]: My bad, I just thought you forgot access modifiers.

**Review by kevinthegreat1 — APPROVED** [review]
Thanks for contributing!

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**